library/src/main/java/com/getstream/sdk/chat/view/Dialog/MoreActionDialog.java

```diff
@@ -56,7 +58,13 @@ public MoreActionDialog setChannelViewModel(ChannelViewModel viewModel) {
 
 
     public MoreActionDialog setMessage(Message message) {
-        this.message = message;
+        // Check Thread parent Message
+        if (viewModel.isThread() && TextUtils.isEmpty(message.getParentId())){
+            this.message = message.copy();
+            this.message.setId(viewModel.getThreadParentMessage().getValue().getId());
```

Comment:
are you sure you need to do this as well? looks very weird that you need to overwrite the message id with the one from the viewmodel

Reply:
Yeah, We **need** this. when switching from main chat to Thread, we had initialized threadparent message's id to empty string.(since some reason as you know)

Reply:
what about we stop touch message.id completely? beside the sending part, the ID is something that should come from API or offline storage.

Reply:
I had to handle it for switching between main channel messages and  Thread messages on only one message list recyclerview.
Got it, Let me think about it.

Reply:
Agree about touching id. And it seems the whole issue is because of cloning entities. Cloning should be a severe exception, because it introduces synchronisation issues. And cloning must be constantly supported.

Reply:
Removed cloning message and surprisingly everything works. @adrian09h could you briefly test the app? Maybe I missed something. If everything is ok will merge it.

Reply:
Will check right now.